src/main/java/filter/expression/QualifierContent.java

```diff
@@ -0,0 +1,66 @@
+package filter.expression;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class QualifierContent {
+
+    //Only one of these will exist in a QualifierContent object
```

Comment:
Isn't it non-trivial to determine the qualifier type based on only the content? For example, `involves` and `authors` use the same value type.

Is it fine to store the `QualifierType` inside `QualifierContent` (or its subtype) class? I found it to be quite odd to implement it that way. If that's the case, won't it contain the same data as the full `Qualifier` class?